### PR TITLE
Merge four working branches SUPINT-1574 NXBT-2989 NXBT-3078 NXBT-3113

### PR DESCRIPTION
**Pull request by jcarsique**
Merged feature-SUPINT-1574-maven-ncp, feature-NXBT-2989-nexus-devtools, feature-NXBT-3078-add-aronder-docker-registry and feature-NXBT-3113-add-staging-repositories
Ready for merge on master

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -22,20 +22,25 @@ VERSION ?= 0.0.0
 
 all: skaffold@up build skaffold@down
 
-build: base builder jenkins central cluster
+build: base builder jenkins central cluster team maven-ncp
```
Comment (jcarsique) [inline Makefile]: jenkins-x

File: Makefile [hunk 2]
```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-..NOTPARALLEL: base builder jenkins central
+..NOTPARALLEL: base builder jenkins-x central cluster team
```
Comment (jcarsique) [inline Makefile]: maven-ncp

File: parms/central/blobstore-parms.json [hunk 3]
```diff
@@ -3,7 +3,7 @@
     "name": "nuxeo-devtools-nexus-central",
     "type": "S3",
     "config": {
-      "bucket": "nuxeo-devtools-nexus-central",
+      "bucket": "nuxeo-devtools-nexus-central-test",
```
Comment (alexistimic) [inline parms/central/blobstore-parms.json]: `nuxeo-devtools-nexus-central` should be the correct blobstore

File: builder/scripts/repository.groovy [hunk 4]
```diff
@@ -1,130 +1,492 @@
...
+    MyException(String message) {
+        this.message = message
+    }
+    String toString() {
+        this.message
+    }
+}
+
+void checkForEmptyValidation(String message, List<String> bad_values) {
+    if(false && bad_values) {
```
Comment (jcarsique) [inline builder/scripts/repository.groovy]: to review

File: postStart.sh [hunk 5]
```diff
@@ -56,19 +58,24 @@ function testLogin() {
...
 if [ -z "${ENABLE_ANONYMOUS_ACCESS}" ]; then
-    secFile="/opt/sonatype/nexus/scripts/security-parms.json"
-    jq '.[] | select (.type == "anonymous") .enabled = "false"' ${secFile} > ${secFile}.tmp && \
-    mv ${secFile}.tmp ${secFile}
+    secFile="$SCRIPTS_PATH/security-parms.json"
+    jq '.[] | select (.type == "anonymous") .enabled = "false"' ${secFile} > ${secFile}.tmp && mv ${secFile}.tmp ${secFile}
 fi
 
-for script in blobstore repository security tasks; do
-    createOrUpdateAndRun ${script} /opt/sonatype/nexus/scripts/${script}-body.json /opt/sonatype/nexus/scripts/${script}-parms.json
```
Comment (jcarsique) [inline postStart.sh]: to later review: execution of tasks.groovy

File: parms/central/blobstore-parms.json (hunk 3 above)
Comment (jcarsique) [reply parms/central/blobstore-parms.json]: the two are correct; `nuxeo-devtools-nexus-central-test` is the bucket for test, set by default
the bucket is renamed for prod by the release pipeline [Jenkinsfile]

**Pull request merged**